core/chains/evm/client/send_only_node.go

```diff
@@ -30,25 +32,34 @@ type SendOnlyNode interface {
 // It only supports sending transactions
 // It must a http(s) url
 type sendOnlyNode struct {
-	uri    url.URL
-	rpc    *rpc.Client
-	geth   *ethclient.Client
-	log    logger.Logger
-	dialed bool
-	name   string
+	uri     url.URL
+	rpc     *rpc.Client
+	geth    *ethclient.Client
+	log     logger.Logger
+	dialed  bool
+	name    string
+	chainID *big.Int
+	ctx     context.Context
+	cancel  context.CancelFunc
 }
 
-func NewSendOnlyNode(lggr logger.Logger, httpuri url.URL, name string) SendOnlyNode {
+// NewSendOnlyNode returns a new sendonly node
+func NewSendOnlyNode(lggr logger.Logger, httpuri url.URL, name string, chainID *big.Int) SendOnlyNode {
 	s := new(sendOnlyNode)
 	s.name = name
 	s.log = lggr.Named("SendOnlyNode").Named(name).With(
 		"nodeTier", "sendonly",
 	)
 	s.uri = httpuri
+	s.chainID = chainID
+	s.ctx, s.cancel = context.WithCancel(context.Background())
 	return s
 }
 
-func (s *sendOnlyNode) Dial(_ context.Context) error {
+// Start setups up and verifies the sendonly node
+// Should only be called once in a node's lifecycle
+// TODO: Failures to dial should put it into a retry loop
```

Comment:
TODO for this PR, or future?

Reply:
Future PR, it's linked somewhere - I think in the test